app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+
+    if user&.authenticate params[:session][:password]
```

Comment:
Toán tử safe navigation `&.` có tác dụng gì?

Reply:
user&.authenticate params[:session][:password] có nghĩa là user && user.authenticate params[:session][:password]
&. là cách viết an toàn để đảm bảo không có các exception không mong muốn bắn ra khi user.authenticate params[:session][:password] bị nil

Reply:
Nếu như user nil thì có chạy hàm đằng sau ko em

Reply:
user nil thì không chạy hàm đằng sau vì (nil && [anything]) == false